File: backend/crawl/app/models/team.py
from sqlalchemy import Column, Integer, String, Text
from sqlalchemy.orm import relationship
from app.models.database import Base


class Team(Base):
    __tablename__ = "team"

    id = Column(Integer, primary_key=True, index=True)
    team_name = Column(String(10))
    team_logo = Column(Text)
    
    players = relationship("Player", back_populates="team")
    home_games = relationship("Game", foreign_keys="Game.home_team_id", back_populates="home_team")
    away_games = relationship("Game", foreign_keys="Game.away_team_id", back_populates="away_team")
    rank = relationship("Rank", uselist=False, back_populates="team")
    team_hitter_stats = relationship("TeamHitterStats", back_populates="team")
    team_pitcher_stats = relationship("TeamPitcherStats", back_populates="team")
    hitter_stats = relationship("HitterStats", back_populates="team")
    pitcher_stats = relationship("PitcherStats", back_populates="team")

File: backend/baseball/src/main/java/com/newbie/baseball/domain/team/entity/Team.java
package com.newbie.baseball.domain.team.entity;

import com.newbie.baseball.domain.game.entity.Game;
import com.newbie.baseball.domain.lineup.entity.LineUp;
import com.newbie.baseball.domain.player.entity.Player;
import com.newbie.baseball.domain.rank.entity.Rank;
import com.newbie.baseball.domain.playerStats.entity.HitterStats;
import com.newbie.baseball.domain.playerStats.entity.PitcherStats;
import com.newbie.baseball.domain.teamStats.entity.TeamHitterStats;
import com.newbie.baseball.domain.teamStats.entity.TeamPitcherStats;
import jakarta.persistence.*;
import lombok.Getter;

import java.util.List;

@Getter
@Entity
@Table(name = "team")
public class Team {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Column(name = "team_name", length = 10)
    private String teamName;

    @Lob
    @Column(name = "team_logo")
    private String teamLogo;

    @OneToMany(mappedBy = "homeTeam")
    private List<Game> homeGames;

    @OneToMany(mappedBy = "awayTeam")
    private List<Game> awayGames;

    @OneToMany(mappedBy = "team")
    private List<Player> players;

    @OneToMany(mappedBy = "team")
    private List<Rank> ranks;

    @OneToMany(mappedBy = "team")
    private List<TeamHitterStats> teamHitterStats;

    @OneToMany(mappedBy = "team")
    private List<TeamPitcherStats> teamPitcherStats;

    @OneToMany(mappedBy = "team")
    private List<HitterStats> hitterStats;

    @OneToMany(mappedBy = "team")
    private List<PitcherStats> pitcherStats;

    @OneToMany(mappedBy = "team")
    private List<LineUp> lineUpList;
}
